kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -263,4 +269,8 @@ private String formatList(List<FileStatus> list) {
     return "\n    "
         + list.stream().map(FileStatus::toString).collect(Collectors.joining(",\n    "));
   }
+
+  public Optional<FileStatus> getLatestChecksum() {
```

Comment:
and I think these needs method docs? This checksum might not be for the `version` that this LogSegment represents, right?

also ... suppose we have 10.checkpoint and 11.json and 12.json ...

what are the possible ranges of the version for which this CRC could be for? Could we have 8.crc? what are the guarantees of this version range?

Reply:
For logSegment, we just have 8.crc, downstream(LogReplay) decides use it or not.